### ISSUE TITLE
preprocess2 fails when topicref is missing regardless of processing-mode

### ISSUE THREAD

**Issue by super-structure** [dita-ot/dita-ot#4454]
## Expected Behavior
For a missing DITA topic file referenced by a `<topicref>` element in a map, the DITA OT should not fail when processing-mode="lax" (defualt) or "skip"; only when set to "strict". This was the behavior in the original preprocess.

A warning should (and is) generated, but the processing should be able to proceed and simply skip over any missing files.

It is not uncommon for a file reference to be removed in one place but missed in others (e.g., a `<xref>` in a topic or `<topicref>` in a relationship table). Providing output for the "lax" or "skip" values allows users to still review the output.

## Actual Behavior
In preprocess2, _regardless_ of the value for processing-mode, the DITA OT fails with a Java NullPointerException. 

## Possible Solution
When the processing-mode is set to "lax" (default) or to "skip", then any files flagged as missing should just be ignored. The warning is already generated in the preprocess2 step of reading the files and generating a temporary copy.

## Steps to Reproduce

1. Include `<topicref href="dummy.dita"/>` within a map of any project; where dummy.dita does not exist.
2. Publish this output to a HTML5 output format with the parameter processing-mode="skip"

## Copy of the error message, log file or stack trace
Relevant portions of the verbose log file:
```
==> Read topic files to temporary store
Reading file:/C:/Users/JASON~1.COL/AppData/Local/Temp/temp20240328103411200/html/639ceaa6fa4030e4017aa08d1da88d9799708894.ditamap
Processing file:/D:/DITA_Dev/example/Sample-Project/map/dummy.dita to file:/C:/Users/JASON~1.COL/AppData/Local/Temp/temp20240328103411200/html/289f519bdf550d83e93b15b4a77bdbbca5af5d06.dita
Error: [DOTX008E] The resource 'file:/D:/DITA_Dev/example/Sample-Project/map/dummy.dita' cannot be loaded. Make sure 'file:/D:/DITA_Dev/example/Sample-Project/map/dummy.dita' exists and that you have permission to access it.
```
```
==> Resolve keyref
Reading file:/C:/Users/JASON~1.COL/AppData/Local/Temp/temp20240328103411200/html/639ceaa6fa4030e4017aa08d1da88d9799708894.ditamap

...

Processing file:/C:/Users/JASON~1.COL/AppData/Local/Temp/temp20240328103411200/html/289f519bdf550d83e93b15b4a77bdbbca5af5d06.dita
Error: Build failed with an exception: D:\DITA\DITA-OT_4.2.2_Dev\plugins\org.dita.base\build.xml:28: The following error occurred while executing this line:
D:\DITA\DITA-OT_4.2.2_Dev\plugins\org.dita.base\build_preprocess2.xml:226: java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "s" is null
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:582)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:440)
	at org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:106)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
	at org.dita.dost.invoker.Main.runBuild(Main.java:853)
	at org.dita.dost.invoker.Main.startAnt(Main.java:240)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
Caused by: D:\DITA\DITA-OT_4.2.2_Dev\plugins\org.dita.base\build_preprocess2.xml:226: java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "s" is null
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:109)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:36)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:437)
	... 17 more
Caused by: java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "s" is null
	at java.base/java.util.Collections.unmodifiableSet(Collections.java:1132)
	at org.dita.dost.writer.KeyrefPaser.getNormalProcessingRoleTargets(KeyrefPaser.java:214)
	at org.dita.dost.module.KeyrefModule.processFile(KeyrefModule.java:520)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.dita.dost.module.KeyrefModule.execute(KeyrefModule.java:152)
	at org.dita.dost.ant.ExtensibleAntInvoker.execute(ExtensibleAntInvoker.java:210)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:299)
	at jdk.internal.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	... 24 more

BUILD FAILED in 3 seconds
```

## Environment

* DITA-OT version: v4.2.2
* Operating system and version:  Windows 11
* How did you run DITA-OT? dita command (via CL) with a project file
* Transformation type: HTML5 (and HTML5-bootstrap)

**Comment by arkadianriver** [comment]
I ran into this, too with 4.2.2, but with no Error found earlier in the log.. just a bunch of info messages about Keyrefs being defined more than once and this same `Cannot invoke "Object.getClass()" because "s" is null` stack trace later referencing `KeyrefParser.java:214`.

Digging around a little, I see PR #4407 on 14 Feb refactors the way some data sets are being passed around. Hoping not to send anyone on a wild-goose chase since I'm not familiar with how the code weaves together, but in the PR the only thing that looks suspect at first glance is `chunk/ChunkModule.java:507` where `toUnmodifiableList()` is changed to `toList()`, which I think allows nulls unlike the former. Maybe somehow a null value is now allowed to make its way to `normalProcessingRoleTargets` that's passed to the `unmodifiableSet()` there on `KeyrefParser.java:214`? Because looking at these last few specific variables and functions mentioned in the stack trace, they haven't changed in years.

I tested on OT 4.2 (11 Feb) and the error doesn't occur.

Hope that helps!
